src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SQLitePersistor.java

```diff
@@ -98,37 +106,40 @@ synchronized void createTable(String tableName, List<Field<?>> columns, String r
                                 String referenceTablePrimaryKeyFieldName) throws SQLException {
     Field foreignKeyField = DSL.field(referenceTablePrimaryKeyFieldName, Integer.class);
     columns.add(foreignKeyField);
-    Table referenceTable = DSL.table(referenceTableName);
-    CreateTableConstraintStep constraintStep = create.createTable(tableName)
-        .column(DSL.field(getPrimaryKeyColumnName(tableName) + PRIMARY_KEY_AUTOINCREMENT_POSTFIX))
-        .columns(columns)
-        .constraints(DSL.constraint(foreignKeyField.getName() + "_FK").foreignKey(foreignKeyField)
-            .references(referenceTable, DSL.field(referenceTablePrimaryKeyFieldName)));
 
-    LOG.debug("table with fk created: {}", constraintStep.toString());
     try {
+      Table referenceTable = DSL.table(referenceTableName);
+      CreateTableConstraintStep constraintStep = create.createTable(tableName)
+          .column(DSL.field(getPrimaryKeyColumnName(tableName) + PRIMARY_KEY_AUTOINCREMENT_POSTFIX))
+          .columns(columns)
+          .constraints(DSL.constraint(foreignKeyField.getName() + "_FK").foreignKey(foreignKeyField)
+              .references(referenceTable, DSL.field(referenceTablePrimaryKeyFieldName)));
       constraintStep.execute();
+      LOG.debug("table with fk created: {}", constraintStep.toString());
       jooqTableColumns.put(tableName, columns);
-    } catch (Exception e) {
-      LOG.error("Failed to create table {}", tableName);
-      throw new SQLException();
+    } catch (DataAccessException e) {
+      String msg = "table " + tableName + " already exists";
+      if (!e.getMessage().contains(msg)) {
+        LOG.error(e);
+        throw new SQLException(e);
+      }
```

Comment:
Since we are consuming the exception, can we add a log statement for the else case that we were not expecting the table to have been created already but we're going to use the one already created?

Reply:
good point !